lib/osf-components/addon/components/cedar-metadata-editor/component.ts

```diff
@@ -9,29 +9,53 @@ import CedarMetadataTemplateModel from 'ember-osf-web/models/cedar-metadata-temp
 import RouterService from '@ember/routing/router-service';
 import { task } from 'ember-concurrency';
 import { waitFor } from '@ember/test-waiters';
+import { action } from '@ember/object';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+import captureException from 'ember-osf-web/utils/capture-exception';
 
 const { cedarConfig } = config;
 
 interface Args {
-    cedarMetadataRecord: CedarMetadataRecordModel;
+    cedarMetadataRecord?: CedarMetadataRecordModel;
     cedarMetadataTemplate: CedarMetadataTemplateModel;
     target: AbstractNodeModel | FileModel;
-    edit: boolean;
+    displayArtifactViewer?: () => {};
 }
 
 export default class CedarMetadataEditor extends Component<Args> {
+    @service toast!: Toast;
+    @service intl!: Intl;
     @service store!: Store;
     @service router!: RouterService;
 
     cedarConfig = cedarConfig.editorConfig;
+    isEdit = this.args.cedarMetadataRecord ? true : false;
+
+    @action
+    addMetadata(): void {
+        if (this.args.cedarMetadataRecord) {
+            const cee = document.querySelector('cedar-embeddable-editor');
+            // eslint-disable-next-line
+            // @ts-ignore
+            cee.instanceObject = this.args.cedarMetadataRecord.metadata;
+        }
+    }
+
+    @action
+    cancel() {
+        if (this.args.displayArtifactViewer) {
+            this.args.displayArtifactViewer();
+        }
+    }
 
     @task
     @waitFor
     async save() {
         const cee = document.querySelector('cedar-embeddable-editor');
         let record: CedarMetadataRecordModel;
-        if (this.args.edit) {
-            record = this.args.cedarMetadataRecord;
+        if (this.isEdit) {
+            record = this.args.cedarMetadataRecord || {} as CedarMetadataRecordModel;
```

Comment:
I think you could just do `this.args.cedarMetadataRecord!` since `if (this.isEdit)` above guarantees that this block is only executed when cedarMetadataRecord is given